HealthcareApp/src/Healthcare.Application/Schedules/Queries/AvailableTimeSlots/GetAvailableSlotsQueryHandler.cs

```diff
@@ -0,0 +1,69 @@
+using Application.Abstractions;
+using Application.Abstractions.Decorators;
+using Domain.Entities;
+using Healthcare.Application.DTOs.Schedule;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+
+namespace Healthcare.Application.Schedules.Queries.AvailableTimeSlots;
+
+public class
+    GetAvailableSlotsQueryHandler : IRequestHandler<GetAvailableSlotsQuery, Dictionary<string, List<TimeSlotDTO>>>
+{
+    private readonly IUserManagerDecorator<ApplicationUser> _userManager;
+    private readonly IScheduleRepository _scheduleRepository;
+
+    public GetAvailableSlotsQueryHandler(IUserManagerDecorator<ApplicationUser> userManager,
+        IScheduleRepository scheduleRepository)
+    {
+        _userManager = userManager;
+        _scheduleRepository = scheduleRepository;
+    }
+
+    public async Task<Dictionary<string, List<TimeSlotDTO>>> Handle(GetAvailableSlotsQuery request,
+        CancellationToken cancellationToken)
+    {
+        var doctors = await _userManager.GetAllUsersAsync();
+        var doctor = await doctors.Include(u => u.DoctorAppointments)
+            .FirstOrDefaultAsync(u => u.Id == request.DoctorId, cancellationToken);
```

Comment:
Encapsulate to repository pattern - you cannot use IQueryable and Includes in your application layer